source/sgp_mode/Instructions.h

```diff
@@ -96,6 +96,25 @@ INST(IO, {
   *a = next;
   state.input_buf.push(next);
 });
+
+INST(Reuptake, {
+  uint32_t next;
+  float score = state.world->GetTaskSet().CheckTasks(state, *a);
+  if (score != 0.0) {
+    state.host->AddPoints(pow(2, score));
+    if (!state.host->IsHost()) {
+      state.world->sym_points_earned += pow(2, score);
+    }
+  }
+  if(state.internalEnvironment->size() > 0){
+    next = (*state.internalEnvironment)[state.internalEnvironment->size() - 1];
+    state.internalEnvironment->pop_back();
+  }
+  else{ next = 0;}
```

Comment:
I'm not sure we want to give them 0 either since that could be pretty useful to them. Can we just not add to the input buffer if there is nothing in the internal environment?